modules/contracts-for-testing/src/main/kotlin/com/r3/corda/lib/tokens/testing/states/Rubles.kt

```diff
@@ -14,38 +14,36 @@ import net.corda.core.transactions.LedgerTransaction
 open class RubleToken(override val amount: Amount<IssuedTokenType<RUB>>,
                       override val holder: AbstractParty) : FungibleToken<RUB>(amount, holder)
 
-open class Ruble : TokenType {
-    override val tokenIdentifier: String
-        get() = "рубль"
-    override val fractionDigits: Int
-        get() = 0
-
-    override fun toString(): String {
-        return "Ruble(tokenIdentifier: ${tokenIdentifier}, fractionDigits: ${fractionDigits})"
+open class Ruble : TokenType("рубль", 0) {
+
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+        if (!super.equals(other)) return false
+        return true
     }
+
+
 }
 
 object RUB : Ruble()
 
 
-open class THING : TokenType {
-    override val tokenIdentifier: String = "PTK"
-    override val fractionDigits: Int = 0
-
-    override fun toString(): String {
-        return "THING(tokenIdentifier: ${tokenIdentifier}, fractionDigits: ${fractionDigits})"
+open class PhoBowl : TokenType("PTK", 0) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+        if (!super.equals(other)) return false
+        return true
     }
 
 }
 
-object PTK : THING()
+object PTK : PhoBowl()
 
 
-data class Appartment(
-        override val tokenIdentifier: String = "FOO"
-) : TokenType {
-    override val fractionDigits: Int = 0
-}
+data class Appartment(val id: String = "Foo") : TokenType(id, 0)
```

Comment:
oh man, not again!!!

Reply:
Haha

Reply:
Apartment only has one "p". How did you miss that @kasiastreich ?!

Reply:
I know how delicate Stefano is, I didn't want to discourage him.

Reply:
Just be glad he didn't get all clever and write it as condomminnnniuuum